cudax/include/cuda/experimental/__device/logical_device.cuh

```diff
@@ -0,0 +1,144 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of CUDA Experimental in CUDA C++ Core Libraries,
+// under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _CUDAX__DEVICE_LOGICAL_DEVICE
+#define _CUDAX__DEVICE_LOGICAL_DEVICE
+
+#include <cuda/__cccl_config>
+
+#if defined(_CCCL_IMPLICIT_SYSTEM_HEADER_GCC)
+#  pragma GCC system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_CLANG)
+#  pragma clang system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_MSVC)
+#  pragma system_header
+#endif // no system header
+
+#include <cuda/experimental/__device/all_devices.cuh>
+#include <cuda/experimental/__green_context/green_ctx.cuh>
+
+namespace cuda::experimental
+{
+namespace detail
+{
+struct logical_device_access;
+}
```

Comment:
I would like to get away from us using `detail` namespace and instead use `__ugly names`